definitions/checks/unread_mail.rb

```diff
@@ -0,0 +1,40 @@
+class Checks::UnreadMail < ForemanMaintain::Check
+  metadata do
+    label :unread_mail
+    description 'Check for unread mail for root user.'
+    tags :default
+  end
+
+  attr_reader :file_path
+
+  def initialize
+    @file_path = '/var/spool/mail/root'
+  end
+
+  def run
+    is_present = file_exists?
+    if is_present
+      unread_mails_count = number_of_mails_present.to_i
+      if unread_mails_count > 0
+        raise_warning(unread_mails_count)
+      end
+    end
+  end
+
+  private
+
+  def raise_warning(mail_count)
+    fail_msg = "WARNING: Found #{mail_count} unread mails for root."
+    fail_msg += ' Please check before proceeding further.'
+    raise ForemanMaintain::Error::Fail, fail_msg
+  end
+
+  def file_exists?
+    File.exist?(file_path)
+  end
+
+  def number_of_mails_present
+    cmd = "egrep '^Message-Id' #{file_path} | wc -l"
```

Comment:
`egrep --count` can provide a count of matching lines, without passing the output to wc for counting.